src/Components/QuickSelect.tsx

```diff
@@ -1,202 +1,226 @@
-                .map((object) => (
-                  <Box>
-                    <Button
-                      onClick={() => handleClick(object)}
-                      color="primary"
-                      variant="text"
-                      size="small"
-                      disableFocusRipple={true}
-                      disableRipple={true}
-                      className={props.classes.quickSelectContainerButton}
-                    ></Button>
-                    <Box mt={-3}>
-                      <Typography color="primary" variant="subtitle2">
-                        {object.join(" ")}
-                      </Typography>
-                    </Box>
-                  </Box>
-                ))}
-            </Box>
-            <Box className={props.classes.flexColumn}>
-              {props.recentlySelected
-                .slice(Math.floor(props.recentlySelected.length / 2))
-                .map((object) => (
-                  <Box>
-                    <Button
-                      onClick={() => handleClick(object)}
-                      color="primary"
-                      variant="text"
-                      size="small"
-                      disableFocusRipple={true}
-                      disableRipple={true}
-                      className={props.classes.quickSelectContainerButton}
-                    ></Button>
-                    <Box mt={-3}>
-                      <Typography color="primary" variant="subtitle2">
-                        {object.join(" ")}
-                      </Typography>
-                    </Box>
-                  </Box>
-                ))}
-            </Box>
-          </Box>
-        </Box>
-      </Box>
-    </Box>
-  );
-};
+import React, { useContext } from "react";
+import { Divider } from "@material-ui/core";
+import { Button, Box, Typography } from "@material-ui/core";
+import { DateSelect } from "./QuickDateSelect";
+import { DateRange, TermContext } from "./DateRange";
+import { GlobalContext } from "./Constants";
+import { DateRangeUI, DropdownData } from "./Types";
+import { MSFormatter } from "./MSFormatter";
+
+interface Inputs {
+  boxClass: string;
+  recentlySelected: number[];
+  dateRange: DateRange;
+  dateRangeUI: DateRangeUI;
+  dropdownData: DropdownData;
+  setDropdownData(data: DropdownData): void;
+  setDateRangeUI(dateRangeUI: DateRangeUI): void;
+  applyChanges(dr: DateRange): void;
+  setRecentlySelected(items: number[]): void;
+}
+
+export const QuickSelect: React.FC<Inputs> = (props) => {
+  const globals = useContext(GlobalContext);
+  let out = props.dropdownData.quickSelectContent;
+
+  let commonlyUsedFirstRow = globals.commonlyUsedText.slice(
+    0,
+    Math.floor(globals.commonlyUsedText.length / 2)
+  );
+
+  let commonlyUsedSecondRow = globals.commonlyUsedText.slice(
+    Math.floor(globals.commonlyUsedText.length / 2),
+    8
+  );
+
+  function getRecentlySelectedFirstRow(): number[] {
```

Comment:
Why are these functions?

Reply:
because the recently selected changes constantly and if I were to abstract it to a variable, it wouldn't update the value, right?

Reply:
Why are they a function instead of being calculated in-line here in the `QuickSelect` function?  They only depend on the props, ergo they can be calculated on execution rather than on invocation.

Reply:
Done